martin/src/bin/martin-cp.rs

```diff
@@ -343,20 +337,53 @@ fn iterate_tiles(tiles: Vec<TileRect>) -> impl Iterator<Item = TileCoord> {
 }
 
 fn check_sources(args: &CopyArgs, state: &ServerState) -> Result<String, MartinCpError> {
-    if let Some(source) = &args.source {
-        Ok(source.clone())
+    if let Some(source_id) = &args.source {
+        Ok(source_id.clone())
     } else {
-        let sources = state.tiles.source_names();
-        if let Some(source) = sources.first() {
-            if sources.len() > 1 {
-                return Err(MartinCpError::MultipleSources(sources.join(", ")));
+        let source_ids = state.tiles.source_names();
+        if let Some(source_id) = source_ids.first() {
+            if source_ids.len() > 1 {
+                return Err(MartinCpError::MultipleSources(source_ids.join(", ")));
             }
-            Ok(source.clone())
+            Ok(source_id.clone())
         } else {
             Err(MartinCpError::NoSources)
         }
     }
 }
+
+fn default_bounds(src: &DynTileSource) -> Vec<Bounds> {
+    if src.sources.is_empty() {
+        vec![Bounds::MAX_TILED]
+    } else {
+        let mut source_bounds = src
+            .sources
+            .iter()
+            .filter_map(|source| source.get_tilejson().bounds)
```

Comment:
🤔 Is this correct?

If a source does not have a bound, I think we need to assume the following

```suggestion
            .map(|source| source.get_tilejson().bounds.unwrap_or_else(|| Bounds::MAX_TILED))
```

what do you think?

Reply:
This would cover a scenario where:
- The user has not specified args for `bbox`
- The user has configured at least one source with bounds (for example, by `config.yml`)
  - If none configured, that is covered by check below after filter map:
  
   ```rs
   if source_bounds.is_empty() {
    info!(
       "No configured bounds for source, using: {}",
       Bounds::MAX_TILED
     );
    vec![Bounds::MAX_TILED]
   ```
- The user has omitted bounds for at least one source (for example, by omitting them in `config.yml`)

---

I had assumed under such a scenario to use just the bounds configured and available, logging to INFO what was being used.

I suppose we could have the secondary fallback here where if _any_ source does not have configured bounds, add to the vec (and thus effective use) the global bounds. The only drawback is it adds another scenario where the tiles for the global bounds need to be processed.

I'll leave it to you to decide which is more intuitive for an inferred default as up to this point all of my use of the `martin-cp` tool has been either very explicit via multiple bbox args/each source's bounds defined in configuration, or just global, depending on the data set I was using it for.

All that said, I don't have a strong preference either way, because either solution still gives the user ample opportunity to specify bounds via args or `config.yml`

Reply:
I think my approach (effectively global bounds) might be more correct for some cases.

Lets prioritise correctness over speed..